deploy/crds/Keycloak.yaml

```diff
@@ -29,6 +29,28 @@ spec:
           type: object
         spec:
           type: object
+          properties:
+            externalDatabaseSecret:
+              description: 'Opotional. A Secret containing the connection string and credentials to an external database. If not provided a Postgres
+                instance will be created in the same namespace'
+              type: string
+            adminCredentialSecret:
```

Comment:
The main issue I have with those references to secrets is how to tell what is the structure of a secret? What fields it should have? Is there any pattern to document things like that?

Reply:
It should be in two places:
1. The CRD (openapi validation section) as @davidffrench pointed out
1. The documentation of the operator